src/libopenrave/kinbodygeometry.cpp

```diff
@@ -563,19 +563,33 @@ inline void LoadJsonValue(const rapidjson::Value& v, KinBody::GeometryInfo::Side
     if(v.IsObject()) {
         orjson::LoadJsonValueByKey(v, "transform", t.transf);
         orjson::LoadJsonValueByKey(v, "halfExtents", t.vExtents);
-        std::string type = "";
-        orjson::LoadJsonValueByKey(v, "type", type);
-        std::map<std::string, KinBody::GeometryInfo::SideWallType> sideWallTypeMapping = {
-            {"nx", KinBody::GeometryInfo::SideWallType::SWT_NX},
-            {"px", KinBody::GeometryInfo::SideWallType::SWT_PX},
-            {"ny", KinBody::GeometryInfo::SideWallType::SWT_NY},
-            {"py", KinBody::GeometryInfo::SideWallType::SWT_PY},
-        };
-
-        if (sideWallTypeMapping.find(type) == sideWallTypeMapping.end()) {
-            throw OPENRAVE_EXCEPTION_FORMAT(_("unrecognized sidewall type %s for loading from json"), type, ORE_InvalidArguments);
+        if(!v.HasMember("type")) {
```

Comment:
@bucketzxm if `type` does not exist, just don't initialize the geometry. perhaps we have geometries like this under normal flow when a user is first creating the geometry. i think throwing exception is too strict...

Reply:
Or just give a default `NX` wall.

Reply:
set default  to `NX`